service/discovery/azure/compute.go

```diff
@@ -436,6 +516,65 @@ func (d *azureComputeDiscovery) discoverVirtualMachines() ([]voc.IsCloudResource
 	return list, nil
 }
 
+func (d *azureComputeDiscovery) handleVirtualMachineScaleSet(vm *armcompute.VirtualMachineScaleSetVM, parent string) (voc.IsCompute, error) {
+	r := &voc.VirtualMachine{
+		Compute: &voc.Compute{
+			Resource: discovery.NewResource(d,
+				voc.ResourceID(*vm.ID),
+				*vm.Name,
+				nil, // not available
+				voc.GeoLocation{
+					Region: *vm.Location,
+				},
+				labels(vm.Tags),
+				voc.ResourceID(parent),
+				voc.VirtualMachineType,
+				vm,
+			),
+		},
+
+		AutomaticUpdates: automaticUpdatesScaleSet(vm.Properties.OSProfile),
```

Comment:
Testing with the testbed on Azure, I get a nil pointer dereference here because `vm.Properties` is nil